plugins/source_docs.go

```diff
@@ -14,20 +15,101 @@ import (
 //go:embed templates/*.go.tpl
 var templatesFS embed.FS
 
+type SourceDocsFormat string
+
+const (
+	SourceDocsFormatMarkdown = "markdown"
+	SourceDocsFormatJSON     = "json"
+)
+
+var SourceDocsFormats = []SourceDocsFormat{
+	SourceDocsFormatMarkdown,
+	SourceDocsFormatJSON,
+}
+
+func (s SourceDocsFormat) String() string {
+	return string(s)
+}
+
+func (s SourceDocsFormat) Validate() error {
+	for _, f := range SourceDocsFormats {
+		if s == f {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid format: %v", s.String())
+}
+
 // GenerateSourcePluginDocs creates table documentation for the source plugin based on its list of tables
-func (p *SourcePlugin) GenerateSourcePluginDocs(dir string) error {
+func (p *SourcePlugin) GenerateSourcePluginDocs(dir string, format SourceDocsFormat) error {
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return err
 	}
 
 	SetDestinationManagedCqColumns(p.Tables())
 
+	switch format {
+	case SourceDocsFormatMarkdown:
+		return p.renderTablesAsMarkdown(dir)
+	case SourceDocsFormatJSON:
+		return p.renderTablesAsJSON(dir)
+	default:
+		return fmt.Errorf("unsupported format: %v", format.String())
+	}
+	return nil
+}
+
+type jsonTable struct {
+	Name        string       `json:"name"`
+	Description string       `json:"description"`
+	Columns     []jsonColumn `json:"columns"`
+	Relations   []jsonTable  `json:"relations"`
+}
+
+type jsonColumn struct {
+	Name        string `json:"name"`
+	Type        string `json:"type"`
+	Description string `json:"description"`
+}
+
+func (p *SourcePlugin) renderTablesAsJSON(dir string) error {
+	tables := p.jsonifyTables(p.Tables())
+	b, err := json.MarshalIndent(tables, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal tables as json: %v", err)
+	}
+	outputPath := filepath.Join(dir, "__tables.json")
```

Comment:
I made this name start with a double-underscore, so that we have the option of dumping a json file for each table in the future without having conflicts with table names.

Reply:
Why not output to stdout?

Reply:
Hmm, that's also an option, but the `doc` command takes a directory as a required argument. So if we do that, we should probably put this under a different command entirely (maybe `json`?)

Reply:
Mmmm, then it only makes sense to write to a file, to not break consistency.